pkg/broker/ingress/server_manager.go

```diff
@@ -39,7 +53,23 @@ func NewServerManager(ctx context.Context, logger *zap.Logger, cmw configmap.Wat
 	httpReceiver := kncloudevents.NewHTTPEventReceiver(httpPort)
 	httpsReceiver := kncloudevents.NewHTTPEventReceiver(httpsPort, kncloudevents.WithTLSConfig(tlsConfig))
 
-	return eventingtls.NewServerManager(ctx, httpReceiver, httpsReceiver, handler, cmw)
+	otelHandler := otelhttp.NewHandler(handler, "broker.ingress",
+		otelhttp.WithMeterProvider(meterProvider),
+		otelhttp.WithTracerProvider(traceProvider),
+		otelhttp.WithFilter(func(r *http.Request) bool {
+			return !network.IsKubeletProbe(r)
+		}),
+		otelhttp.WithPropagators(tracing.DefaultTextMapPropagator()),
+		otelhttp.WithSpanNameFormatter(func(operation string, r *http.Request) string {
+			if r.URL.Path == "" {
+				return r.Method + " /"
+			}
+			return fmt.Sprintf("%s %s", r.Method, r.URL.Path)
+		}),
+
+	)
```

Comment:
This seems very similar to the code in `filter`.  It would be great to have a common library function for setting this up, unless there are differences I'm missing.